### PR TITLE
Add background and dryrun function

### PR DESCRIPTION
**Pull request by chiahaoliu**
Add `background` and `dryrun` function and include `dryrun` option to `_unpack_and_run`. Code passed both simulation and `unittest`.

I still leave `get_light_image_dryrun` and other similar dryrun function in in `xpdaca.py`. If this refactoring is okay, I will clean them up in next commit.

### CODE REVIEW

File: xpdacq/xpdacq.py [hunk 1]
```diff
@@ -257,6 +257,48 @@ def calibration(sample, scanplan, auto_dark = glbl.auto_dark, **kwargs):
     _execute_scans(scan, auto_dark, auto_calibration = False, light_frame = True, dryrun = False)
     return
 
+def background(sample, scanplan, auto_dark = glbl.auto_dark, **kwargs):
```
Comment (sbillinge) [inline xpdacq/xpdacq.py]: Check if it is better to do it this way or to do ``auto_dark = None`` then inside the function
```
if auto_dark == None:
    auto_dark = glbl.auto_dark
```
I don't mind which way it is, but I thought that @tacaswell  liked it better this way....

File: xpdacq/xpdacq.py [hunk 2]
```diff
@@ -287,54 +329,45 @@ def dark(sample, scanplan, **kwargs):
     _yamify_dark(dark_def)
     return dark_uid
 
-def dryrun(sample,scan,**kwargs):
-    '''same as run but scans are not executed.
+def background(sample, scanplan, **kwargs):
```
Comment (sbillinge) [inline xpdacq/xpdacq.py]: The function name and the docstring do not seem to agree with each other.  name suggests a background scan but docstring a dryrun

File: xpdacq/xpdacq.py (hunk 1 above)
Comment (chiahaoliu) [inline xpdacq/xpdacq.py]: :+1:

File: xpdacq/xpdacq.py (hunk 2 above)
Comment (chiahaoliu) [inline xpdacq/xpdacq.py]: This was corrected in a commit later than this one.

**Comment by chiahaoliu** [comment]
`auto_dark` issue has been fixed in last commit.

**Pull request merged**